types/aws-lambda/trigger/appsync-resolver.d.ts

```diff
@@ -0,0 +1,44 @@
+import { Handler } from '../handler';
+
+export type AppSyncResolverHandler<T, V> = Handler<AppSyncResolverEvent<T>, V | V[]>;
+
+/**
+ * See https://docs.aws.amazon.com/appsync/latest/devguide/resolver-context-reference.html
+ *
+ * @param T type of the arguments
+ */
+export interface AppSyncResolverEvent<T> {
+    arguments: T;
+    identity?: AppSyncIdentityIAM | AppSyncIdentityCognito;
+    source?: { [key: string]: any };
+    request: {
+        headers: { [key: string]: string };
+    };
+    info: {
+        selectionSetList: string[];
+        selectionSetGraphQL: string;
+        parentTypeName: string;
+        fieldName: string;
```

Comment:
I'm multiplexing multiple resolvers into the same handler by checking these fields, which means that in the type I use these are also type parameters. E.g. (simplified):

```ts
type MyEvent =
  | AppSyncResolverEvent<"Query", "post", { id: string }, Post>
  | AppSyncResolverEvent<"Mutation", "addPost", PostInput, Post>
  | ...;
```

Reply:
I am not sure how this would look like in the code. I am not really happy with loads of generic args imho